python/lsst/obs/base/gen2to3/calibRepoConverter.py

```diff
@@ -0,0 +1,158 @@
+
+if TYPE_CHECKING:
+    from lsst.daf.butler import StorageClass
+    from ..cameraMapper import CameraMapper
+    from ..mapping import Mapping as CameraMapperMapping  # disambiguate from collections.abc.Mapping
+    from .filePathParser import FilePathParser
+
+CURATED_CALIBRATION_DATASET_TYPES = (
+    "defects",
+    "camera",
+    "transmission_sensor",
+    "transmission_filter",
+    "transmission_optics",
+    "transmission_atmosphere",
+    "bfKernel"
+)
+
+
+class CalibRepoConverter(RepoConverter):
+    """A specialization of `RepoConverter` for calibration repositories.
+
+    Parameters
+    ----------
+    mapper : `CameraMapper`
+        Gen2 mapper for the data repository.  The root associated with the
+        mapper is ignored and need not match the root of the repository.
+    kwds
+        Additional keyword arguments are forwarded to (and required by)
+        `RepoConverter`.
+    """
+
+    def __init__(self, *, mapper: CameraMapper, **kwds):
+        super().__init__(**kwds)
+        self.mapper = mapper
+        self._datasetTypes = []
+
+    def isDatasetTypeSpecial(self, datasetTypeName: str) -> bool:
+        # Docstring inherited from RepoConverter.
+        return datasetTypeName in CURATED_CALIBRATION_DATASET_TYPES
+
+    def isDirectorySpecial(self, subdirectory: str) -> bool:
+        # Docstring inherited from RepoConverter.
+        return False
+
+    def iterMappings(self) -> Iterator[Tuple[str, CameraMapperMapping]]:
+        # Docstring inherited from RepoConverter.
+        yield from self.mapper.calibrations.items()
+
+    def makeDataIdExtractor(self, datasetTypeName: str, parser: FilePathParser,
+                            storageClass: StorageClass) -> DataIdExtractor:
+        # Docstring inherited from RepoConverter.
+        extractor = DataIdExtractor(
+            datasetTypeName,
+            storageClass,
+            filePathParser=parser,
+            universe=self.task.universe,
+            instrument=self.task.instrument.getName(),
+        )
+        self._datasetTypes.append(extractor.datasetType)
+        return extractor
+
+    def insertDimensionData(self):
+        # Docstring inherited from RepoConverter.
+        # TODO: refactor this into a retargetable subtask, so we can customize
+        # it for different obs packages.  The catch is that it needs to
+        # generate calibration_label strings consistent with those produced
+        # by the Translator system, and that's harder to refactor into
+        # something retargetable.
+        db = sqlite3.connect(os.path.join(self.root, "calibRegistry.sqlite3"))
+        db.row_factory = sqlite3.Row
+        records = []
+        for datasetType in self._datasetTypes:
+            if "calibration_label" not in datasetType.dimensions:
+                continue
+            fields = ["validStart", "validEnd", "calibDate"]
+            if "detector" in datasetType.dimensions.names:
+                fields.append("ccd")
```

Comment:
:+1: I'll add a `ccdKey` config option here too, then.  That's at least a step in the right direction.

Do you know if other cameras also use `calibDate`/`validStart`/`validEnd` (with just `calibDate` appearing in the file timeplates)?  If so, this could already be pretty generic.

Reply:
They do all seem to have to calibDate/validStart/validEnd keys and calibData in the templates (except obs_cfht which doesn't seem to have any of those).

Reply:
Great.  I'll back off in the ambition of the to-do comment above then, as it sounds like that would be overkill.  Hopefully a bit of testing and tweaking with other cameras will suffice.